userguide/tutorials/invoice_examples.adoc

```diff
@@ -0,0 +1,132 @@
+= Invoice Examples
+
+This document provides examples of different invoice scenarios. 
+
+== Prerequisites
+
+Ensure that you have gone through the https://docs.killbill.io/latest/userguide_subscription.html#components-invoicing[_Subscription User Guide_] and are familiar with the basic invoicing concepts.
+
+
+== Invoice Terminology - work in progress
+
+Whenever there is a change in the system (a recurring subscription billing, an external charge, a change in plan, and so on), the system generates an invoice. An invoice consists of one or more invoice items of different types. In addition, an invoice consists of several components like `balance`, `chargedAmount`, `paidAmount` and so on. Let us first understand some of these terms. 
+
+
+* *Invoice Item*: Represents an amount corresponding to a type of charge, credit or adjustment. The different types of invoice items are `RECURRING`, `FIXED`, `EXTERNAL_CHARGE`, `USAGE`, `TAX`, `ITEM_ADJ`, `CREDIT_ADJ`, `REPAIR_ADJ` and `CBA_ADJ` as  explained in the https://docs.killbill.io/latest/userguide_subscription.html#components-invoicing[subscription billing] document. 
+
+* *Invoice Payment*: Represents a payment corresponding to an invoice. It has a `type` and an `amount`. The different types of invoice payments are `ATTEMPT` (represents successful payment), `CHARGED_BACK` (represents a chargeback) and `REFUND` (represents a refund) 
+
+
+* *chargedAmount*: The amount that the customer is charged. It is computed by adding up the amounts on invoice items of type `FIXED`, `RECURRING`, `EXTERNAL_CHARGE`, `REPAIR_ADJ`, `ITEM_ADJ`, `USAGE` and `TAX`. So, `chargedAmount=FIXED + RECURRING + EXTERNAL_CHARGE + REPAIR_ADJ + ITEM_ADJ + USAGE + TAX`  
+//TODO, add about including CREDIT_ADJ in case of DRAFT invoice?
+
+* *amountPaid*: The amount that the customer has paid. Is calculated by subtracting invoice payments of type `REFUND` and `CHARGEBACK` from invoice payments of type `ATTEMPT`. In other words, `amountPaid=attempt - (refund + chargeback)`
+
+* *invoiceBalance*: The amount that the customer owes or zero (in case the invoice corresponds to a credit).  It is calculated by subtracting credits, adjustments and the `amountPaid` from the `chargedAmount`. In other words, invoiceBalance = chargedAmount - amountPaid -amountRefunded - SUM(CBA_ADJ)  - SUM(ITEM_ADJ). The invoice balance is always 0 if the invoice is in a `DRAFT` or `VOID` status or if it is writtenoff, is a migration invoice or has a zero parent invoice balance (in case the invoice corresponds to a child account). 
```

Comment:
> The amount that the customer owes or zero (in case the invoice corresponds to a credit). 

This is true. I would suggest slightly different though:

Positive amount if the customer owes something or zero otherwise.

Perhaps defer the invoice credit discussion after all these definitions, and then have a small 'Note' talking about invoice credit and redefine what `invoiceBalance`, ... are for those (and later provide example)

Reply:
Fixed the part about `invoice balance`. Also moved the invoice credit explanation to the invoice credit section.